### PR TITLE
Feature/hw 2

### PR DESCRIPTION
**Pull request by SlavikSurminskiy**
1) Added general layout styles (header, footer, main content)
2) Created courses page

### CODE REVIEW

File: src/app/breadcrumbs/breadcrumbs.component.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.breadcrumbs-wrapper {
+  padding: 20px;
+  background: #fff;
```
Comment (roksolanachipko) [inline src/app/breadcrumbs/breadcrumbs.component.scss]: Can you create variables for all colors and move to shared file?
(Also if we have some standard fonts, sizes, paddings and so on)

File: src/app/course-card/course-card.component.scss [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  }
+
+  &__actions {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-end;
+    flex-grow: 1;
+    flex-shrink: 0;
+    flex-basis: 200px;
```
Comment (roksolanachipko) [inline src/app/course-card/course-card.component.scss]: Can we make it?
```suggestion
    flex: 1 0 200px;
```

File: src/app/course-card/course-card.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
+import { ICourse } from '../shared/models/course.model';
```
Comment (roksolanachipko) [inline src/app/course-card/course-card.component.ts:2]: Why naming is ICourse?

File: src/app/breadcrumbs/breadcrumbs.component.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-breadcrumbs',
+  templateUrl: './breadcrumbs.component.html',
+  styleUrls: ['./breadcrumbs.component.scss']
+})
+export class BreadcrumbsComponent implements OnInit {
+
+  constructor() { }
```
Comment (roksolanachipko) [inline src/app/breadcrumbs/breadcrumbs.component.ts]: We can delete constructor, if it's empty

File: src/app/course-card/course-card.component.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    title: '',
+    creationDate: '',
+    duration: 0,
+    description: '',
+  };
+
+  @Output() edit: EventEmitter<string> = new EventEmitter();
+  @Output() delete: EventEmitter<string> = new EventEmitter();
+
+  constructor() { }
```
Comment (roksolanachipko) [inline src/app/course-card/course-card.component.ts]: We can delete constructor, if it's empty

File: src/app/course/course.component.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-course',
+  templateUrl: './course.component.html',
+  styleUrls: ['./course.component.scss']
+})
+export class CourseComponent implements OnInit {
+
+  constructor() { }
```
Comment (roksolanachipko) [inline src/app/course/course.component.ts]: We can delete constructor, if it's empty
Same for all construtors

File: src/app/course/course.component.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+  ngOnInit(): void { }
```
Comment (roksolanachipko) [inline src/app/course/course.component.ts]: It seams like ngOnInit also is not needed here

File: src/app/courses/courses.component.html [hunk 8]
```diff
@@ -1 +1,35 @@
...
+  </div>
+
+  <div class="courses__add">
+    <button mat-raised-button color="accent">
+      <mat-icon>add</mat-icon> Add
+    </button>
+  </div>
+
+  <app-course-card
+    *ngFor="let c of courses"
```
Comment (roksolanachipko) [inline src/app/courses/courses.component.html]: let's use full names
```suggestion
    *ngFor="let course of courses"
```

File: src/app/courses/courses.component.ts [hunk 9]
```diff
@@ -1,12 +1,37 @@
...
+  courses: ICourse[] = [];
+
+  constructor(private coursesService: CoursesService) {}
+
+  ngOnInit(): void {
+    this.courses = this.coursesService.getCourses();
+  }
+
+  onSearch(): void {
+    console.log(this.searchQuery);
```
Comment (roksolanachipko) [inline src/app/courses/courses.component.ts:23]: do we need this console.log?
Please check whole file

**Review by roksolanachipko — APPROVED** [review]

File: src/app/course-card/course-card.component.ts (hunk 3 above)
Comment (SlavikSurminskiy) [reply src/app/course-card/course-card.component.ts:2]: Usually I add "I"(interface) prefix to all interfaces.

File: src/app/courses/courses.component.ts (hunk 9 above)
Comment (SlavikSurminskiy) [reply src/app/courses/courses.component.ts:23]: According to homework we should log @Output event from component to console and log search query as well

File: src/app/course/course.component.ts (hunk 6 above)
Comment (SlavikSurminskiy) [reply src/app/course/course.component.ts]: Ok, I'll delete all empty constructors

File: src/app/course-card/course-card.component.scss (hunk 2 above)
Comment (SlavikSurminskiy) [reply src/app/course-card/course-card.component.scss]: Good catch, thanks

File: src/app/breadcrumbs/breadcrumbs.component.scss (hunk 1 above)
Comment (SlavikSurminskiy) [reply src/app/breadcrumbs/breadcrumbs.component.scss]: Got it

File: src/app/courses/courses.component.html (hunk 8 above)
Comment (SlavikSurminskiy) [reply src/app/courses/courses.component.html]: Fixed

**Pull request merged**